Quotient/events/roommessageevent.cpp

```diff
@@ -140,16 +140,22 @@ RoomMessageEvent::RoomMessageEvent(const QJsonObject& obj)
     if (content.contains(MsgTypeKey) && content.contains(BodyKey)) {
         auto msgtype = content[MsgTypeKey].toString();
         bool msgTypeFound = false;
-        for (const auto& mt : msgTypes)
+        for (const auto& mt : msgTypes) {
             if (mt.matrixType == msgtype) {
                 _content = mt.maker(content);
                 msgTypeFound = true;
             }
+        }
 
         if (!msgTypeFound) {
             qCWarning(EVENTS) << "RoomMessageEvent: unknown msg_type,"
                               << " full content dump follows";
             qCWarning(EVENTS) << formatJson << content;
+            return;
+        }
+
+        if (content.contains(RelatesToKey)) {
+            _relatesTo = fromJson<std::optional<EventRelation>>(content[RelatesToKey]);
         }
```

Comment:
```suggestion
            fromJson(content[RelatesToKey], _relatesTo);
```
`fillFromJson<std::optional<>>` is intelligent enough to not even touch the target structure if the key is missing; but I'd rather make sure to initialise `_relatesTo`, hence using `fromJson()`.